pkg/process/metadata/parser/java/jee.go

```diff
@@ -0,0 +1,253 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package javaparser contains functions to autodetect service name for java applications
+package javaparser
+
+import (
+	"archive/zip"
+	"encoding/xml"
+	"errors"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/afero/zipfs"
+)
+
+// appserver is an enumeration of application server types
+type serverVendor uint8
+
+// appserver enums
+const (
+	unknown serverVendor = 0
+	jboss                = 1 << (iota - 1)
+	tomcat
+	weblogic
+	websphere
+)
+
+const (
+	// app servers hints
+	wlsServerMainClass  string = "weblogic.Server"
+	wlsHomeSysProp      string = "-Dwls.home="
+	websphereJar        string = "ws-server.jar"
+	websphereMainClass  string = "defaultServer"
+	tomcatMainClass     string = "org.apache.catalina.startup.Bootstrap"
+	tomcatSysProp       string = "-Dcatalina.base="
+	jbossStandaloneMain string = "org.jboss.as.standalone"
+	jbossDomainMain     string = "org.jboss.as.server"
+	jbossSysProp        string = "-Djboss.home.dir="
+	applicationXMLPath  string = "/META-INF/application.xml"
+)
+
+var (
+	errUnhandledDeployment = errors.New("unhandled deployment type")
+)
+
+type (
+	// applicationXML is used to unmarshal information from a standard EAR's application.xml
+	// example doc: https://docs.oracle.com/cd/E13222_01/wls/docs61/programming/app_xml.html
+	applicationXML struct {
+		XMLName     xml.Name `xml:"application"`
+		ContextRoot []string `xml:"module>web>context-root"`
+	}
+	// deployedAppFindFn is used to find the application deployed on a domainHome
+	// args should be supplied since some vendors may require additional information from them (i.e. server name)
+	deployedAppFindFn func(domainHome string, args []string, fs afero.Fs) ([]string, bool)
+	// warContextRootFindFn is used to extract the context root from a vendor defined configuration inside the war.
+	// if not found it returns en empty string and false
+	warContextRootFindFn func(fs afero.Fs) (string, bool)
+	// defaultWarContextRootFn returns the default naming that apply for a certain fileName.
+	// it is usually the file without the extension, but it can differ for some vendors (i.e. tomcat)
+	defaultWarContextRootFn func(fileName string) string
+)
+
+// definitions of standard extractors
+var (
+	deploymentFinders = map[serverVendor]deployedAppFindFn{
+		weblogic: weblogicFindDeployedApps,
+	}
+	contextRootFinders = map[serverVendor]warContextRootFindFn{
+		weblogic: weblogicExtractWarContextRoot,
+	}
+	defaultContextNameExtractors = map[serverVendor]defaultWarContextRootFn{
+		weblogic: standardExtractContextFromWarName,
+	}
+)
```

Comment:
do we have more than weblogic entries? do we plan to have more?

Reply:
yes for this reasons I put in place this logic (strongly inspired from service.go :) ). Each app server will have its own logic to parse deployed add a vendor specific way to declare the context root in war deployments. Also, i.e. for tomcat, the default naming for war may change(i.e. weblogic uses the war filename without the ext but tomcat strips other information). At term, we'll have logic for all the declared vendors (websphere, jboss, tomcat and jetty)